mdoc-js/src/main/scala/mdoc/modifiers/JsModifier.scala

```diff
@@ -173,15 +176,37 @@ class JsModifier extends mdoc.PreModifier {
           val outjsfile = resolveOutputJsFile(inputFile)
           outjsfile.write(new String(content))
           val outmdoc = outjsfile.resolveSibling(_ => "mdoc.js")
-          outmdoc.write(Resources.readPath("/mdoc.js"))
+          val selectMocTemplate = config.moduleKind match {
+            case CommonJSModule => "/mdoc_nomodule.js"
+            case ESModule => "/mdoc_esmodule.js"
+            case NoModule => "/mdoc_nomodule.js"
+          }
+          outmdoc.write(Resources.readPath(selectMocTemplate))
           val relfile = outjsfile.toRelativeLinkFrom(ctx.outputFile, config.relativeLinkPrefix)
           val relmdoc = outmdoc.toRelativeLinkFrom(ctx.outputFile, config.relativeLinkPrefix)
-          new CodeBuilder()
-            .println(config.htmlHeader)
-            .lines(config.libraryScripts(outjsfile, ctx))
-            .println(s"""<script type="text/javascript" src="$relfile" defer></script>""")
-            .println(s"""<script type="text/javascript" src="$relmdoc" defer></script>""")
-            .toString
+          config.moduleKind match {
+            case CommonJSModule =>
+              new CodeBuilder()
+                .println(config.htmlHeader)
+                .lines(config.libraryScripts(outjsfile, ctx))
+                .println(s"""<script type="text/javascript" src="$relfile" defer></script>""")
+                .println(s"""<script type="text/javascript" src="$relmdoc" defer></script>""")
+                .toString
+            case ESModule =>
+              new CodeBuilder()
+                .println(config.htmlHeader)
+                .lines(config.libraryScripts(outjsfile, ctx))
+                .println(s"""<script type="module" src="$relfile"></script>""")
+                .println(s"""<script type="module" src="$relmdoc"></script>""")
+                .toString
+            case NoModule =>
+              new CodeBuilder()
```

Comment:
Could we deduplicate this with the first case since it's the same?

Reply:
Yes. I wasn't sure if it was more readable to split the cases into the three cases, or whether it would be better to have an if... I can change it.

Reply:
We could even just do `case CommonJSModule | NoModule =>` which would be simplest